EIPS/eip-2645.md

```diff
@@ -0,0 +1,68 @@
+---
+eip: 2645
+title: Hierarchical Deterministic Wallet for Computation Integrity Proof (CIP) Layer-2
+author: Tom Brand (tom@starkware.co), Louis Guthmann (louis@starkware.co)
+discussions-to: https://ethereum-magicians.org/t/hierarchical-deterministic-wallet-for-computation-integrity-proof-cip-layer-2/4286
+status: Draft
+type: Standards Track 
+category: ERC
+created: 2020-05-13
+---
+
+## Simple Summary
+In the context of Computation Integrity Proof (CIP) Layer-2 solutions such as ZK-Rollups, users are required to sign messages on new elliptic curves optimized for those environnements. These curves are Baby Jubjub (in the context of the BN254 SNARK approach) and Arya in (the context of StarkEx). Extensive work has been done to provide secure ways to derive private keys. We leverage this work to define an efficient way to securely derive private keys from existing infrastructure, as well as creating domain separation between Layer-2 applications.
+
+## Abstract
+We provide a Derivation Path allowing a user to derive hierarchical keys for Layer-2 solutions depending on the zk-technology, the application, the user’s Layer-1 address, as well as an efficient grinding method to enforce the private key distribution within the curve domain.
+
+## Motivation
+In the context of Computation Integrity Proof (CIP) Layer-2 solutions such as ZK-Rollups, users are required to sign messages on new elliptic curves optimized for those environnements. Extensive work has been done to make it secure on Bitcoin via [BIP32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki), [BIP39](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki) and [BIP44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki). Those protosals have become the standard for wallets in the entire industry, independently on the associated blockchain. As Layer-2 solutions are taking off, it was a necessary requirement to maintain the same standard and security in this new space. StarkWare, in the context of the launch of their first products, needed to explore the existing solutions and adapt them to match its requirements. While working with wallets, it felt that the effort needed to be shared to benefit the community and get additional feedback. This present EIP is the result of this work. 
```

Comment:
> StarkWare, in the context of the launch of their first products, needed to explore the existing solutions and adapt them to match its requirements. While working with wallets, it felt that the effort needed to be shared to benefit the community and get additional feedback. This present EIP is the result of this work.

We have been having problems with people using EIPs as a means of advertising their products, so we generally prefer to not mention/reference external products in EIPs unless it is critical to the EIP.

Recommend removing these last few sentences, as they are more history than motivation.  Alternatively, consider rewording this to be less historic and more present-tense motivation.

In general, the motivation section should target an audience of someone 5-10 years from now looking at this EIP and wondering why they might want to implement it.  Listing potential use cases/applications are a great example of this.

Reply:
Removed all mention of Starkware